tools/gce-change-disktype/main.py

```diff
@@ -0,0 +1,277 @@
+  body = {
+      'name': snapshot_name,
+  }
+  logging.debug('Creating snapshot of disk project=%s, zone=%s, disk=%s',
+                project, zone, disk)
+  operation = compute.disks().createSnapshot(
+      project=project, zone=zone, disk=disk, body=body).execute()
+  result = wait_for_zonal_operation(compute, project, zone, operation)
+  logging.debug('Snapshotting of disk project=%s, zone=%s, disk=%s complete.',
+                project, zone, disk)
+  return result
+
+
+def delete_snapshot(compute, project, snapshot_name):
+  logging.debug('Deleting snapshot project=%s, snapshot_name=%s', project,
+                snapshot_name)
+  operation = compute.snapshots().delete(
+      project=project, snapshot=snapshot_name).execute()
+  result = wait_for_global_operation(compute, project, operation)
+  logging.debug('Deleting snapshot project=%s,  snapshot_name=%s complete.',
+                project, snapshot_name)
+  return result
+
+
+def attach_disk(compute, project, zone, instance, disk, boot, auto_delete, deviceName):
+  """ Attaches disk to instance.
+
+  Requries iam.serviceAccountUser
+  """
+  disk_url = 'projects/{0}/zones/{1}/disks/{2}'.format(project, zone, disk)
+  body = {
+      'autoDelete': auto_delete,
+      'boot': boot,
+      'deviceName' : deviceName,
+      'source': disk_url,
+  }
+  logging.debug('Attaching disk project=%s, zone=%s, instance=%s, disk=%s',
+                project, zone, instance, disk_url)
+  operation = compute.instances().attachDisk(
+      project=project, zone=zone, instance=instance, body=body).execute()
+  result = wait_for_zonal_operation(compute, project, zone, operation)
+  logging.debug(
+      'Attaching disk project=%s, zone=%s, instance=%s, disk=%s complete.',
+      project, zone, instance, disk_url)
+  return result
+
+
+def detach_disk(compute, project, zone, instance, disk):
+  logging.debug('Detaching disk project=%s, zone=%s, instance=%s, disk=%s',
+                project, zone, instance, disk)
+  operation = compute.instances().detachDisk(
+      project=project, zone=zone, instance=instance, deviceName=disk).execute()
+  result = wait_for_zonal_operation(compute, project, zone, operation)
+  logging.debug(
+      'Detaching disk project=%s, zone=%s, instance=%s, disk=%s complete.',
+      project, zone, instance, disk)
+  return result
+
+
+def delete_disk(compute, project, zone, disk):
+  logging.debug('Deleting disk project=%s, zone=%s, disk=%s', project, zone,
+                disk)
+  operation = compute.disks().delete(
+      project=project, zone=zone, disk=disk).execute()
+  result = wait_for_zonal_operation(compute, project, zone, operation)
+  logging.debug('Deleting disk project=%s, zone=%s, disk=%s complete.', project,
+                zone, disk)
+  return result
+
+def create_disk(compute, project, region, zone, snapshot_name, disk_name,
+                disk_type):
+  """Creates a new CMEK encrypted persistent disk from a snapshot"""
```

Comment:
There is some leftover stuff from the original script

Reply:
Cleaned it up.